pkg/resources/sku/sku.go

```diff
@@ -0,0 +1,157 @@
+package sku
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/integr8ly/integreatly-operator/apis/v1alpha1"
+	appsv1 "github.com/openshift/api/apps/v1"
+	appsv12 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v13 "k8s.io/api/core/v1"
+	"reflect"
+)
+
+type SKU struct {
+	name           string
+	productConfigs map[v1alpha1.ProductName]ProductConfig
+	isUpdated      bool
+}
+
+type ProductConfig struct {
+	productName     v1alpha1.ProductName
+	resourceConfigs map[string]ResourceConfig
+	sku             *SKU
+}
+
+type ResourceConfig struct {
+	Replicas  int32                       `json:"replicas,omitempty"`
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+type RateLimit struct {
+	Unit            string  `json:"unit,omitempty"`
+	RequestsPerUnit int64   `json:"requests_per_unit,omitempty"`
+	AlertLimits     []int64 `json:"alert_limits,omitempty"`
+}
+
+type skuConfigReceiver struct {
+	Name      string                    `json:"name,omitempty"`
+	RateLimit RateLimit                 `json:"rate-limiting,omitempty"`
+	Resources map[string]ResourceConfig `json:"resources,omitempty"`
+}
+
+func GetSKU(SKUId string, SKUConfig *corev1.ConfigMap, retSku *SKU) error {
+	received := &[]skuConfigReceiver{}
```

Comment:
Could we rename this `allSKUs` it took me a few moments to work out why we have this as well as `skuReceiver`